ambari-server/src/main/java/org/apache/ambari/server/state/ServiceComponentImpl.java

```diff
@@ -468,8 +468,8 @@ public void debugDump(StringBuilder sb) {
     for (ServiceComponentHost sch : hostComponents.values()) {
       if (!first) {
         sb.append(" , ");
-        first = false;
       }
+      first = false;
```

Comment:
This change was not really necessary... Now first will be set to false for every iteration, not just for the one case where first was previously true.

Reply:
Quite the opposite.  It was set to `false` only if it was already `false`, ie. never.

Reply:
Yikes. Thanks for correcting me.